pkg/monitoring/event/types.go

```diff
@@ -0,0 +1,80 @@
+package event
+
+import (
+	"crypto/sha256"
+
+	bin "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+)
+
+var (
+	SetConfigDiscriminator       []byte
+	SetBillingDiscriminator      []byte
+	RoundRequestedDiscriminator  []byte
+	NewTransmissionDiscriminator []byte
+)
+
+func init() {
+	sumSetConfig := sha256.Sum256([]byte("event:SetConfig"))
+	SetConfigDiscriminator = sumSetConfig[:8]
+
+	sumSetBilling := sha256.Sum256([]byte("event:SetBilling"))
+	SetBillingDiscriminator = sumSetBilling[:8]
+
+	sumRoundRequested := sha256.Sum256([]byte("event:RoundRequested"))
+	RoundRequestedDiscriminator = sumRoundRequested[:8]
+
+	sumNewTransmission := sha256.Sum256([]byte("event:NewTransmission"))
+	NewTransmissionDiscriminator = sumNewTransmission[:8]
```

Comment:
Maybe a helper func could dedupe and allow these to be set inline intsead?

Reply:
I found this difficult to read, and was struggling to see why it needs to exist inside a `func init` at all, rather than be set inline where the variable is declared.

I'm not at all concerned about overhead during initialization. That happens either way.

If this compiles, then IMHO it is already much easier to understand:
```go
	SetConfigDiscriminator       = sha256.Sum256([]byte("event:SetConfig"))[:8]
```
If this can't be inlined trivially, or is still difficult to read, then calling a simple helper func that accepts just the string would make it simultaneously readable and testable:
```go
	SetConfigDiscriminator       = someHelperFunc("event:SetConfig")
```

Reply:
Aha, that makes sense. I added a helper and refactored